src/java/org/apache/cassandra/index/sai/utils/PrimaryKey.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.index.sai.utils;
+
+import org.apache.cassandra.db.Clustering;
+import org.apache.cassandra.db.ClusteringComparator;
+import org.apache.cassandra.db.DecoratedKey;
+import org.apache.cassandra.dht.Token;
+import org.apache.cassandra.utils.bytecomparable.ByteComparable;
+import org.apache.cassandra.utils.bytecomparable.ByteSource;
+
+/**
+ * Representation of the primary key for a row consisting of the {@link DecoratedKey} and
+ * {@link Clustering} associated with a {@link org.apache.cassandra.db.rows.Row}.
+ *
+ * For legacy V1 support only the {@link DecoratedKey} will ever be supported for a row.
+ *
+ * For the V2 on-disk format the {@link DecoratedKey} and {@link Clustering} are supported.
```

Comment:
Are we going to have those legacy primary keys here?

Reply:
No, I will remove these 2 lines of comment because they make no sense for a memory only index.